File: Core/src/main/java/com/darwinreforged/server/core/events/util/Listener.java
package com.darwinreforged.server.core.events.util;

import java.lang.annotation.*;

/**
 Annotation for methods indicating the method should receive events.

 @author Matt */
@Documented
@Inherited  // @Inherited doesn't work for method, so we have to implement that manually anyway...
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Listener {
    /**
     Listener priority. Lower values are called first.

     @return Priority of event receiving.
     */
    Priority value() default Priority.NORMAL;
}

File: core/src/main/kotlin/org/dockbox/darwin/core/annotations/Listener.kt
package org.dockbox.darwin.core.annotations

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class Listener(val value: Priority = Priority.NORMAL) {
    enum class Priority(val priority: Int) {
        LAST(0x14), LATE(0xf), NORMAL(0xa), EARLY(0x5), FIRST(0x0);
    }
}
